cmake/Dependencies.cmake

```diff
@@ -1093,8 +1093,8 @@ if(USE_ROCM)
     hip_include_directories(${Caffe2_HIP_INCLUDE})
 
     set(Caffe2_PUBLIC_HIP_DEPENDENCY_LIBS
-      ${PYTORCH_HIP_LIBRARIES} ${PYTORCH_MIOPEN_LIBRARIES} ${hipcub_LIBRARIES} ${ROCM_HIPRTC_LIB} ${ROCM_ROCTX_LIB})
-    list(APPEND Caffe2_PUBLIC_HIP_DEPENDENCY_LIBS ${hipblaslt_LIBRARIES})
+      hip::amdhip64 MIOpen hiprtc::hiprtc) # libroctx will be linked in with MIOpen
```

Comment:
@naromero77amd I'd also mention roctx as a direct dependency, even if it's pulled in as a dependency of MIOpen, because PyTorch does depend on roctx, independent of its dependency on MIOpen? Also, if the only usage of variables like `${PYTORCH_MIOPEN_LIBRARIES` was here, can we clean up the code that defines them in LoadHip.cmake? (another one is `PYTORCH_RCCL_LIBRARIES`, which is used in [rccl.cmake](https://github.com/search?q=repo%3Apytorch%2Fpytorch%20PYTORCH_RCCL_LIBRARIES&type=code))

Reply:
Capturing offline discussion: this PR will limit itself to changes to address https://github.com/pytorch/pytorch/issues/131701, and a future PR will address further cleanup.